packages/apollo-codegen-typescript/src/codeGeneration.ts

```diff
@@ -246,6 +335,68 @@ export class TypescriptAPIGenerator extends TypescriptGenerator {
     this.scopeStackPop();
   }
 
+  public getGlobalTypesUsedForOperation(doc: Operation | Fragment, context: CompilerContext) {
+    let docTypesUsed: GraphQLType[] = [];
+
+    if (doc.hasOwnProperty('operationName')) {
+      const operation = doc as Operation;
+      docTypesUsed = operation.variables.map(({ type }) => type);
+    }
+
+    const reduceTypesForDocument = (
+      nested: SelectionSet,
+      acc: GraphQLType[]
+    ) => {
+      acc = nested.selections
```

Comment:
Instead of setting and immediately returning `acc`, can you just directly return it here?